File: leetcode/0069-sqrtx/Solution.java
class Solution {
    public int mySqrt(int x) {
        long lo = 0, hi = x;
        
        while (lo < hi) {
            long mid = (lo + hi + 1) / 2;
            
            if (mid * mid <= x)
                lo = mid;
            else
                hi = mid - 1;
        }
        
        return (int)lo;
    }
}

File: leetcode/0069-sqrtx/solution.py
class Solution:
    def mySqrt(self, x: int) -> int:
        lo, hi = 0, x
        
        while lo < hi:
            mid = lo + (hi - lo + 1) // 2
            
            if mid * mid <= x:
                lo = mid
            else:
                hi = mid - 1
        
        return lo